bundle/libraries/same_name_libraries.go

```diff
@@ -0,0 +1,72 @@
+package libraries
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type checkForSameNameLibraries struct{}
+
+var patterns = []dyn.Pattern{
+	taskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	forEachTaskLibrariesPattern.Append(dyn.AnyIndex(), dyn.AnyKey()),
+	envDepsPattern.Append(dyn.AnyIndex()),
+}
+
+func (c checkForSameNameLibraries) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	var diags diag.Diagnostics
+	libBaseNames := make(map[string]bool)
+
+	err := b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
+		var err error
+		for _, pattern := range patterns {
+			v, err = dyn.MapByPattern(v, pattern, func(p dyn.Path, lv dyn.Value) (dyn.Value, error) {
+				libPath := lv.MustString()
+				// If not local library, skip the check
+				if !IsLibraryLocal(libPath) {
+					return lv, nil
+				}
+
+				lib := filepath.Base(lv.MustString())
+				if libBaseNames[lib] {
```

Comment:
If I read this correctly, this will trigger an error if you have 2 tasks that both use the same library.

This is fine if they all use the same full path. It is only an error if two or more use a different full path but have the same basename. This is the conflict we need to error on.

Reply:
Thanks good point! Fixed and added a test case